network/common/src/main/java/org/apache/spark/network/server/OneForOneStreamManager.java

```diff
@@ -109,15 +111,34 @@ public void connectionTerminated(Channel channel) {
     }
   }
 
+  @Override
+  public void checkAuthorization(TransportClient client, long streamId) {
+    if (client.getClientId() != null) {
```

Comment:
That would be true, but I don't believe you can actually set things up like that. Authentication is either enabled on the server or its not, for all clients.